### PR TITLE
Traefik, Airflow, OpenWhisk

### PR DESCRIPTION
**Pull request by gsemet**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,3 +14,10 @@ To populate zk with the ansible vault (only need to be done once, or when new in
     cat 'XXX' > ~/.vault-password.forestscribe  # get this file from pierre
     export ZK_HOST=<zk_host>:<zk_port>
     ansible-playbook zk_populate.yml --vault-password-file=~/.vault-password.forestscribe
+
+For Traefik related:
+
+    export DOMAIN_EXT=<domainserver.ext>
```
Comment (tardyp) [inline README.md]: put this is the vault

File: group_vars/all [hunk 2]
```diff
@@ -62,14 +63,31 @@ servicePorts:
...
+    traefik_https:            30302
+
+    api_gateway:              30310
+
 
 
 zk_host: "{{ ansible_env.ZK_HOST }}"
 zookeeper_uri: "zk://{{ zk_host }}"
 zookeeper_etc_uri: "{{ zookeeper_uri }}/etc"
+domain_ext: "{{ ansible_env.DOMAIN_EXT }}"
```
Comment (tardyp) [inline group_vars/all]:
```
+domain_ext: "{{ zk.cloud.DOMAIN_EXT }}
```

File: marathon/airflow/celery.yml [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+          env:
+            EXECUTOR: "Celery"
+            POSTGRES_USER: airflow
+            POSTGRES_PASSWORD: airflow
+            POSTGRES_HOST: "{{ docker_ip }}"
+            POSTGRES_PORT: "{{ servicePorts.airflow_pg }}"
+            POSTGRES_DB: "airflow"
+            REDIS_HOST: "{{ docker_ip }}"
+            REDIS_PORT: "{{ servicePorts.airflow_redis }}"
+            FRENET_KEY: AFKp0LrregZGOucp22eOr0jW0Ru2PmbZzBWoeCdL8Vk=
```
Comment (tardyp) [inline marathon/airflow/celery.yml]: change and put in the zk vault

**Review by tardyp — CHANGES_REQUESTED** [review]

File: marathon/airflow/celery.yml [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+            http_proxy: http://proxy-ir.{{ domain_ext }}:911
```
Comment (tardyp) [inline marathon/airflow/celery.yml]: there are already proxy config
```
                https_proxy: "{{zk.proxy_env.https_proxy}}"
                http_proxy: "{{zk.proxy_env.http_proxy}}"
                no_proxy: "{{zk.proxy_env.no_proxy}}"
                ALL_PROXY: "{{zk.proxy_env.ALL_PROXY}}"
```

File: marathon/airflow/celery.yml [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+          disk: 0
+          instances: 5
+          args: ["worker"]
+          container:
+            type: DOCKER
+            volumes:
+              - hostPath: airflowdags
+                containerPath: /usr/local/airflow/dags
+                mode: RW
+              - containerPath: "airflowdags"
```
Comment (tardyp) [inline marathon/airflow/celery.yml]: this really needs to be stored here?
This shall be stored on git right?

File: marathon/core/traefik.yml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                  image: stibbons31/traefik-workaround-marathon-bad-gateway:workaround_marathon_bad_gateway
+                  # network: "HOST"
+                  network: "BRIDGE"
+                  privileged: true
+                  parameters: "{{ docker_log_driver_parameters }}"
+                  forcePullImage: false
+                  portMappings:
+                    - containerPort: 80
+                      protocol: tcp
+                      hostPort: 80
```
Comment (tardyp) [inline marathon/core/traefik.yml]: don't put hostPort if network is bridge

File: marathon/core/traefik.yml [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+                    - containerPort: 443
+                      protocol: tcp
+                      hostPort: 443
+                      name: "https"
+                      servicePort: "{{ servicePorts.traefik_https }}"
+                    - containerPort: 19050
+                      hostPort: 19050
+                      protocol: tcp
+                      name: "admin"
+                      servicePort: "{{ servicePorts.traefik_admin }}"
```
Comment (tardyp) [inline marathon/core/traefik.yml]: I think this should got to traefik.forestscribe.xxx.com

File: marathon/core/traefik.yml [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+                volumes:
+                  - containerPath: "/log"
+                    hostPath: "log"
+                    mode: "RW"
+              env:
+                http_proxy: http://proxy-ir.{{ domain_ext }}:911
```
Comment (tardyp) [inline marathon/core/traefik.yml]: use zk config

File: marathon/core/traefik.yml [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+                  #  timeoutSeconds: 20
+                  maxConsecutiveFailures: 3
+                  #    ignoreHttp1xx: false
+              # residency:
+              #   taskLostBehavior: "WAIT_FOREVER"
+              # upgradeStrategy:
+              #   minimumHealthCapacity: 0.5
+              #   maximumOverCapacity: 0
+              labels:
+                subdomain: 't'
```
Comment (tardyp) [inline marathon/core/traefik.yml]: why not put subdomain to forestscribe?

**Review by tardyp — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (gsemet) [reply README.md]: done

File: marathon/airflow/celery.yml (hunk 3 above)
Comment (gsemet) [reply marathon/airflow/celery.yml]: done

File: marathon/airflow/celery.yml (hunk 5 above)
Comment (gsemet) [reply marathon/airflow/celery.yml]: maybe a shared volume ?

File: marathon/airflow/celery.yml (hunk 4 above)
Comment (gsemet) [reply marathon/airflow/celery.yml]: done

File: marathon/core/traefik.yml (hunk 6 above)
Comment (gsemet) [reply marathon/core/traefik.yml]: needed to force to take the port 80 (it's it the entrypoint for routing) and have the URL in the marathon ui with the right port.

File: marathon/core/traefik.yml (hunk 7 above)
Comment (gsemet) [reply marathon/core/traefik.yml]: it can do it, just need to deal with the port that is not 80

File: marathon/core/traefik.yml (hunk 8 above)
Comment (gsemet) [reply marathon/core/traefik.yml]: done

File: marathon/core/traefik.yml (hunk 9 above)
Comment (gsemet) [reply marathon/core/traefik.yml]: done

File: marathon/airflow/celery.yml (hunk 5 above)
Comment (tardyp) [reply marathon/airflow/celery.yml]: with buildbot, we have a http clone from github to load the master.cfg

File: marathon/core/traefik.yml (hunk 6 above)
Comment (tardyp) [reply marathon/core/traefik.yml]: user host mode

**Comment by tardyp** [comment]
looks like you said done, but you didn't push

**Pull request merged**